### PR TITLE
quantum_teleportation.py

### PR DESCRIPTION
**Pull request by KevinJoven11**
This code is for the #hacktoberfest
This circuit run the teleportation circuit using Qiskit. You can change the teleported state as preference.

### Describe your change:
I add the quantum teleportation algorithm using qiskit. This is my first Pull Request let me know if I did it well.
Best,

* [ X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ X] This pull request is all my own work -- I have not plagiarized.
* [ X] I know that pull requests will not be merged if they fail the automated tests.
* [ X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ X] All new Python files are placed inside an existing directory.
* [ X] All filenames are in all lowercase characters with no spaces or dashes.
* [ X] All functions and variable names follow Python naming conventions.
* [ X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: quantum/quantum_teleportation.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    # Build the circuit
+    qc.u(theta, phi, lam, 0)  # Quantum State to teleport
+    qc.h(1)
+    qc.cx(1, 2)
+    qc.cx(0, 1)
+    qc.h(0)
+    qc.cz(0, 2)
+    qc.cx(1, 2)
+
+    qc.measure([2], [0])
```
Comment (CaedenPH) [inline quantum/quantum_teleportation.py:54]:
```suggestion
    quantum_circuit = QuantumCircuit(qr, cr)  # Define the quantum circuit.

    # Build the circuit
    quantum_circuit.u(theta, phi, lam, 0)  # Quantum State to teleport
    quantum_circuit.h(1)
    quantum_circuit.cx(1, 2)
    quantum_circuit.cx(0, 1)
    quantum_circuit.h(0)
    quantum_circuit.cz(0, 2)
    quantum_circuit.cx(1, 2)

    quantum_circuit.measure([2], [0])
```
I'm mulling over whether or not I prefer this. Just an idea though, go with what you prefer

File: quantum/quantum_teleportation.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    # Simulate the circuit usong qasm simulator
+    backend = Aer.get_backend("qasm_simulator")
+    job = execute(qc, backend, shots=1000)
+
+    return job.result().get_counts(qc)
+
+
+if __name__ == "__main__":
+    print(
+        f"Total count for teleported state is: {quantum_teleportation(np.pi/2, np.pi/2, np.pi/2)}"
```
[the remark this answers is not in the document]
Comment (KevinJoven11) [reply quantum/quantum_teleportation.py:65]: Perfect!

File: quantum/quantum_teleportation.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+https://qiskit.org/textbook/ch-algorithms/teleportation.html
+"""
+
+import numpy as np
+from qiskit import Aer, ClassicalRegister, QuantumCircuit, QuantumRegister, execute
+
+
+def quantum_teleportation(
+    theta: float = np.pi / 2, phi: float = np.pi / 2, lam: float = np.pi / 2
+):
```
[the remark this answers is not in the document]
Comment (KevinJoven11) [reply quantum/quantum_teleportation.py:19]: This return the counts probabilities of the qubit teleported.

[the remark this answers is not in the document]
Comment (CaedenPH) [reply quantum/quantum_teleportation.py:19]: @KevinJoven11 Then indicate that with the python return typehint. 

For example:
```py
def add(a: int, b: int) -> int:
    return a + b
```

[the remark this answers is not in the document]
Comment (KevinJoven11) [reply quantum/quantum_teleportation.py]: Thanks! I think that I did it in one of the pull request. Name "python return typehint solved".

File: quantum/quantum_teleportation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cclauss) [reply quantum/quantum_teleportation.py:68]: Relying only on the default values sounds boring to me.  We want to test multiple sets of values.

**Review by cclauss — APPROVED** [review]

**Pull request merged**